### ISSUE TITLE
Text field loses focus when message is received

### ISSUE THREAD

**Issue by exuvo** [thomotron/Phinix#51]
### Description
When i am in the process of writing a response if someone else send a message my text field loses focus.

### Steps to reproduce
1. Focus text field and start writing
2. Someone else sends a chat message
3. Your text field has now lost focus and your button presses do random shit as they are hotkeys for game functions.

**Comment by thomotron** [comment]
I'm not actually 100% sure what causes this. I'd done some testing with it a little while ago to see if it was some kind of in-game event or something but it wasn't comclusive.

Sometimes I'd be able to go a couple of minutes without any issues and other times it'd be near-constantly losing focus.

**Comment by exuvo** [comment]
Nice work on redoing the UI!
As a workaround would it be possible to re-take focus if it was held before a message was received? Like keep a bool variable around with if the chat field was focused last frame and force focus the chat input field when a new message is received?

**Comment by thomotron** [comment]
Possibly? I've stumbled across [GUI.FocusControl()](https://docs.unity3d.com/ScriptReference/GUI.FocusControl.html) and [EditorGUI.FocusTextInControl()](https://docs.unity3d.com/ScriptReference/EditorGUI.FocusTextInControl.html) which might get what we're after. Might give these a try some time this week and see if there's any success.

**Comment by thomotron** [comment]
I've drilled into this one as best I can and while I've managed to find one of the causes. It seemed as though Unity was stealing the focus when drawing the highlight behind the chat messages, so I've swapped it out with a different method that doesn't do that.

As for when chat messages were coming in, I truly have no idea. From what I could tell it happens ~130-140 frames after the chat is updated, so I'm not 100% convinced it's coming from the chat list itself, but debugging it further turned into a nightmare so I've just resorted to aggressively keeping focus on the field until you click out of it.
